nexus/tests/integration_tests/metrics.rs

```diff
@@ -296,10 +296,40 @@ async fn test_timeseries_schema_list(
         .expect("Failed to find HTTP request latency histogram schema");
 }
 
-pub async fn timeseries_query(
+/// Run an OxQL query until it succeeds or panics.
+pub async fn timeseries_query_until_success(
     cptestctx: &ControlPlaneTestContext<omicron_nexus::Server>,
     query: impl ToString,
 ) -> Vec<oxql_types::Table> {
+    let query_ = query.to_string();
+    wait_for_condition(
+        || async {
+            match timeseries_query(cptestctx, &query_).await {
+                Some(r) => Ok(r),
+                None => Err(CondCheckError::<()>::NotYet),
+            }
+        },
+        &Duration::from_secs(1),
+        &Duration::from_secs(30),
+    )
+    .await
+    .unwrap_or_else(|_| {
+        panic!(
+            "Timeseries named in query are not available, query: '{}'",
```

Comment:
Take it or leave it: maybe it's worth including the maximum wait duration in this panic message? I might turn it into a constant so we can format the actual value rather than just saying "after 30s" in the format string.

Reply:
Done in 910a9695b